ogc/src/main/scala/geotrellis/server/ogc/params/ExtendedParameters.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.server.ogc.params
+
+import geotrellis.raster.TargetCell
+import com.azavea.maml.ast.{Expression, FocalHillshade, FocalSlope}
+
+import cats.Semigroup
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.Valid
+import cats.syntax.option._
+import cats.syntax.apply._
+import cats.instances.option._
+import cats.syntax.semigroup._
+
+/**
+ * A case class with extended parameters that can be defined for the GeoTrellis server endpoints.
+ *
+ * @param azimuth  azimuth for the FocalSlope and FocalHillshade configuration
+ * @param altitude altitude the FocalSlope and FocalHillshade configuration
+ * @param zFactor  zFactor for the FocalSlope and FocalHillshade configuration
+ * @param target   the TargetCell for the Focal operation
+ */
+case class ExtendedParameters(azimuth: Option[Double], altitude: Option[Double], zFactor: Option[Double], target: Option[TargetCell]) {
+  // override the default semigroup behavior
+  implicit val doubleSemigroup: Semigroup[Double] = new Semigroup[Double] {
+    def combine(l: Double, r: Double): Double = l
+  }
+
+  def bindExpression(expr: Expression): Expression = {
+    def deepMap(expression: Expression, f: Expression => Expression): Expression =
+      f(expression).withChildren(expression.children.map(f).map(deepMap(_, f)))
+
+    deepMap(expr, {
+      case e @ FocalHillshade(_, az, al, z, t) => e.copy(
```

Comment:
The complexity is in this `map` function that should know how to use user defined parameters.